### DISCUSSION TITLE
env-vars not available in v1.39.0

### DISCUSSION THREAD

**Discussion by antonneubauer** [mobile-dev-inc/Maestro#2100] [General]
**TLDR**: The following fails with 1.39.0, passes with 1.38.1. The command is part of a job executed on a gitlab runner using docker executor. The tests fail because env vars are not read properly even though they are available in the job environment.

Command:
```bash
maestro --device $EMULATOR_DEVICE_ID test --format html --output "MAESTRO_OUTPUT/maestro-report-$CI_COMMIT_SHORT_SHA.html" --debug-output MAESTRO_OUTPUT --flatten-debug-output tests/maestro/flows
```

----

Since the release of 1.39.0 my flows don't pass anymore because values that were read from environment variables are not available in flows anymore.

They are set and available in the pipeline job (echo-ing them right before maestro execution results in the expected behaviour). The vars contain text that is entered in the flows and looking at the maestro log "null" is entered instead.

I am fairly certain the problem arose with the new release because another pipeline job, with identical setup, runs in parallel using Maestro v1.38.1 and works perfectly fine. When moving the failing job back to v1.38.1 it passes again.

Is anyone else experiencing this? For now I have just fixed the version number but in general I am interested in staying up to date on maestro versions.

**Comment by amanjeetsingh150** [comment]
Hey @antonneubauer thank for reaching out! Which type of environment variable exactly are not available?

https://maestro.mobile.dev/api-reference/configuration/workspace-configuration

Mentioned here? Or the one exported in shell?

If you could create a GitHub issue that can help reproducing problem locally.

**Reply by antonneubauer** [reply]
Hey @amanjeetsingh150 
the varibales are exported to the shell environment. I am running the tests locally, though, not using `maestro cloud ...` but `maestro test ...`. I am accessing them as specified here: https://maestro.mobile.dev/advanced/parameters-and-constants#accessing-variables-from-the-shell

Currently I don't really have the time to set up a minimal test case and unfortunately cannot give you access to my setup. Sorry.

**Reply by Fishbowler** [reply]
Do they begin `MAESTRO_?
They'll need to for Maestro to read them. Else Maestro would be slurping all sorts of things it has no business accessing.
